### PR TITLE
feat: add governance prop to support earn deposits from community pool

### PR DESCRIPTION
**Pull request by karzak**
Adds `CommunityPoolDepositProposal` type, which supports deposits from the community pool to earn vaults via governance proposal.

### CODE REVIEW

File: proto/kava/earn/v1beta1/proposal.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/kava-labs/kava/x/earn/types";
+
+// CommunityPoolDepositProposal deposits from the community pool into an earn vault
+message CommunityPoolDepositProposal {
+  option (gogoproto.goproto_stringer) = false;
+  option (gogoproto.goproto_getters)  = false;
+
+  string                       title          = 1 [(gogoproto.moretags) = "yaml:\"title\""];
```
Comment (drklee3) [inline proto/kava/earn/v1beta1/proposal.proto:14]: I believe we don't need the yaml tags anymore, the `gogoproto.moretag` options can be removed

**Review by drklee3 — APPROVED** [review]

File: x/earn/keeper/proposal_handler.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/earn/types"
+)
+
+// HandleCommunityPoolDepositProposal is a handler for executing a passed community pool deposit proposal
+func HandleCommunityPoolDepositProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolDepositProposal) error {
+	// deposit from community pool address (kava1jv65s3grqf6v6jl3dp4t6c9t9rk99cd8m2splc)
+	return k.Deposit(ctx, k.accountKeeper.GetModuleAddress("distribution"), p.Amount)
```
Comment (rhuairahrighairidh) [inline x/earn/keeper/proposal_handler.go:12]: I beleive we'll need to update the community pool object stored in `distribution` like in `DistributeFromFeePool`: https://github.com/cosmos/cosmos-sdk/blob/v0.45.6/x/distribution/keeper/fee_pool.go.

The distribution module account holds the community pool plus pending staking rewards, so if we don't update it we could deposit too much and prevent user's from being able to claim.

File: x/earn/types/proposal.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+func (cdp CommunityPoolDepositProposal) String() string {
+
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf(`Community Pool Multi Spend Proposal:
+  Title:       %s
+  Description: %s
+  Amount:   %s
+`, cdp.Title, cdp.Description, cdp.Amount))
+	return b.String()
+}
```
Comment (rhuairahrighairidh) [inline x/earn/types/proposal.go:57]: nit: I think `String` methods aren't necessary with proto as they're generated

**Comment by karzak** [comment]
closing in favor of #1297

**Pull request closed without merging**